### PR TITLE
add init nodes that are called before channel subhandle

### PR DESCRIPTION
**Pull request by quentinql**

### CODE REVIEW

**Review by Zluurk — APPROVED** [review]

File: pypeman/tests/test_channel.py [hunk 1]
```diff
@@ -39,6 +41,11 @@ def raise_exc(msg):
     raise Exception()
 
 
+def return_text(msg, text="toto"):
```
Comment (klausfmh) [inline pypeman/tests/test_channel.py:44]: whay default value for text?

Why value toto?

File: pypeman/tests/test_channel.py [hunk 2]
```diff
@@ -234,6 +241,28 @@ def test_final_nodes(self):
             vars(self.clean_msg(msg1)), vars(self.clean_msg(endnode_input)),
             "Channel final_nodes don't takes event msg in input")
 
+    def test_init_nodes(self):
+        """ Whether BaseChannel init_nodes is working """
+        chan1 = BaseChannel(name="test_channel_init_clbk", loop=self.loop)
+        initouttext = "tutu"
```
Comment (klausfmh) [inline pypeman/tests/test_channel.py:247]: why tutu and not just an empty text or "unspecified" r "default"?
Is this a relict from debugging?

File: pypeman/tests/test_channel.py [hunk 3]
```diff
@@ -234,6 +241,28 @@ def test_final_nodes(self):
...
+        chan1.add(n1)
+        msg1 = generate_msg(message_content="startmsg")
+        self.start_channels()
+
+        self.loop.run_until_complete(chan1.handle(msg1))
+
+        n1_input = n1.last_input()
+        self.assertEqual(
+            n1_input.payload, initouttext,
+            "Channel init_nodes doesn't seems to work")
```
Comment (klausfmh) [inline pypeman/tests/test_channel.py:264]: doesn't seem to work

**Review by klausfmh — APPROVED** [review]

File: pypeman/tests/test_channel.py (hunk 2 above)
Comment (quentinql) [reply pypeman/tests/test_channel.py]: Yup relict from debug that i didnt change, changed to inittxt

File: pypeman/tests/test_channel.py (hunk 3 above)
Comment (quentinql) [reply pypeman/tests/test_channel.py]: done

**Pull request merged**